src/ui/public/query_bar/directive/query_bar.html

```diff
@@ -51,6 +56,9 @@
           class="kuiLocalSearchInput"
           ng-class="{'kuiLocalSearchInput-isInvalid': queryBarForm.$invalid}"
           data-test-subj="queryInput"
+          aria-autocomplete="list"
+          aria-controls="query-bar-suggestions-typeahead-items"
+          aria-activedescendant="{{queryBar.focusedTypeaheadItemID}}"
```

Comment:
For whatever reason you need to set `role="textbox"` on those two `input` elements - even though it implicitally already has that row according to a11y tree debugging. Otherwise NVDA on windows doesn't seem to have the autocompletion properly working if it's missing.